cocos2d/motion-streak/CCMotionStreak.js

```diff
@@ -26,508 +24,247 @@
  * streak will move before adding a new ribbon segment, and the texture                     <br/>
  * length is the how many pixels the texture is stretched across. The texture               <br/>
  * is vertically aligned along the streak segment.
- * @class
- * @extends _ccsg.Node
- *
- * @property {cc.Texture2D} texture                         - Texture used for the motion streak.
- * @property {Boolean}      fastMode                        - Indicate whether use fast mode.
- * @property {Boolean}      startingPositionInitialized     - Indicate whether starting position initialized.
- * @example
- * //example
- * new cc.MotionStreak(2, 3, 32, cc.Color.GREEN, s_streak);
+ * !#zh 运动轨迹，用于游戏对象的运动轨迹上实现拖尾渐隐效果。
+ * @class MotionStreak
+ * @extends Component
  */
-cc.MotionStreak = _ccsg.Node.extend(/** @lends cc.MotionStreak# */{
-    texture:null,
-    fastMode:false,
-    startingPositionInitialized:false,
-
-    _blendFunc:null,
-
-    _stroke:0,
-    _fadeDelta:0,
-    _minSeg:0,
-
-    _maxPoints:0,
-    _nuPoints:0,
-    _previousNuPoints:0,
-
-    /* Pointers */
-    _pointVertexes:null,
-    _pointState:null,
-
-    // webgl
-    _vertices:null,
-    _colorPointer:null,
-    _texCoords:null,
-
-    _verticesBuffer:null,
-    _colorPointerBuffer:null,
-    _texCoordsBuffer:null,
-    _className:"MotionStreak",
-
-    /**
-     * creates and initializes a motion streak with fade in seconds, minimum segments, stroke's width, color, texture filename or texture   <br/>
-     * Constructor of cc.MotionStreak
-     * @param {Number} fade time to fade
-     * @param {Number} minSeg minimum segment size
-     * @param {Number} stroke stroke's width
-     * @param {Number} color
-     * @param {string|cc.Texture2D} texture texture filename or texture
-     */
-    ctor: function (fade, minSeg, stroke, color, texture) {
-        _ccsg.Node.prototype.ctor.call(this);
-        this._positionR = cc.p(0, 0);
-        this._blendFunc = new cc.BlendFunc(cc.macro.SRC_ALPHA, cc.macro.ONE_MINUS_SRC_ALPHA);
-
-        this.fastMode = false;
-        this.startingPositionInitialized = false;
-
-        this.texture = null;
-
-        this._stroke = 0;
-        this._fadeDelta = 0;
-        this._minSeg = 0;
-
-        this._maxPoints = 0;
-        this._nuPoints = 0;
-        this._previousNuPoints = 0;
-
-        /** Pointers */
-        this._pointVertexes = null;
-        this._pointState = null;
-
-        // webgl
-        this._vertices = null;
-        this._colorPointer = null;
-        this._texCoords = null;
-
-        this._verticesBuffer = null;
-        this._colorPointerBuffer = null;
-        this._texCoordsBuffer = null;
-
-        if(texture !== undefined)
-            this.initWithFade(fade, minSeg, stroke, color, texture);
+var MotionStreak = cc.Class({
+    name: 'cc.MotionStreak',
+
+    // 1.由于拖尾自身创建的贴图是基于局部坐标点的，所以需要创建一个父节点，来改变它为全局坐标点。
+    // 2.由于当移动父节点的时候，无法得知自身是否移动，所以需要自身更新自身的全局坐标。
```

Comment:
To avoid conflict with other render component, we haven't use ComponentUnderSG, its implementation also requires some different approach:

1. Needed a parent node to make motion streak's position global related
2. Need to update the position in each frame by itself because we don't know whether the global position have changed